test/unit/lodash.spec.js

```diff
@@ -0,0 +1,709 @@
+            expect(_.assign.apply(_, params)).to.have.ownProperty('number', 6);
+            expect(_.assign.apply(_, params)).to.have.ownProperty('square', 25);
+            expect(_.assign.apply(_, params)).to.have.ownProperty('string', "Trust me! I am a engineer!")
+            expect(_.assign.apply(_, params).ping()).to.eql('Number: 6');
+            expect(_.assign.apply(_, params).pong()).to.eql('Number: 25');
+        });
+
+    });
+
+    describe('assignIn', function () {
+
+        var params;
+        beforeEach(function(){
+            params = datasets.assignIn();
+        });
+
+        it('should check types ', function(){
+            var elem1 = params[0];
+            var elem2 = params[1];
+
+            expect(typeof elem1).to.eql('object');
+            expect(elem2 instanceof Object).to.eql(true);
+        });
+
+        it('should check properties obj1', function(){
+            expect(_.assign(params[0])).to.have.ownProperty('number', 5);
+            expect(_.assign(params[0])).to.have.ownProperty('square');
+            expect(params[0].square()).to.eql('Square: ' + Math.pow(Math, this.number));
+        });
+
+        it('should check properties obj2', function(){
+            expect(_.assign(params[1])).to.have.ownProperty('number', 4);
+            expect(_.assign(params[1])).to.have.ownProperty('cube', 64);
+            expect(_.assign(params[1])).to.have.property('square');
+            expect(_.assign(params[1])).not.to.have.ownProperty('square');
+            expect(params[1].square()).to.eql('Square: ' + Math.pow(Math, this.number));
+        });
+        
+        it('should check if obj2 have prototype properties', function(){
+            expect(_.assign(params[1])).not.to.have.ownProperty('fourthPower', 256);
+            expect(_.assign(params[1])).to.have.property('fourthPower', 256);
+        });
+
+        it('should return new object with properties with both objects', function(){
+            expect(_.assignIn.apply(_, params)).to.have.ownProperty('number', 4);
+            expect(_.assignIn.apply(_, params)).to.have.ownProperty('cube', 64);
+            expect(_.assignIn.apply(_, params)).to.have.ownProperty('fourthPower', 256);
+            expect(_.assignIn.apply(_, params).square()).to.eql('Square: ' + Math.pow(Math, this.number));
+        });
+    });
+
+    describe('findKey', function () {
+        describe('findKey1', function () {
+
+            var params;
+            beforeEach(function(){
+                params = datasets.findKey1();
+            });
+            it('should check types', function(){
+                var elem1 = params[0];
+                var elem2 = params[1];
+
+                expect(elem2 instanceof Function).to.eql(true);
+            });
+
+
+            it('should return person who has number greather than 20', function(){
+                var obj1 = {};
+
+                _.times(100,function(index) {
```

Comment:
You are enforcing knowledge of other lodash function to solve this one. I think usage is good, however we shouldn't do that for simple lodash exercises I think.